Subject: [patch] uml fix update_process_times call.patch
From: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
Patch-mainline: 2.6.11-rc1

In call to update_process_times() set parameter user
correctly. (was from for SKAS).

Signed-off-by: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
Signed-off-by: Jeff Dike <jdike@addtoit.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 arch/um/kernel/time_kern.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN arch/um/kernel/time_kern.c~uml-fix-update_process_times-call arch/um/kernel/time_kern.c
--- 25/arch/um/kernel/time_kern.c~uml-fix-update_process_times-call	Fri Dec  3 13:50:41 2004
+++ 25-akpm/arch/um/kernel/time_kern.c	Fri Dec  3 13:50:41 2004
@@ -170,7 +170,7 @@ void __const_udelay(um_udelay_t usecs)
 void timer_handler(int sig, union uml_pt_regs *regs)
 {
 	local_irq_disable();
-	update_process_times(user_context(UPT_SP(regs)));
+	update_process_times(CHOOSE_MODE(user_context(UPT_SP(regs)), (regs)->skas.is_user));
 	local_irq_enable();
 	if(current_thread->cpu == 0)
 		timer_irq(regs);
_
